Packs/DeHashed/Integrations/DeHashed/DeHashed.py

```diff
@@ -0,0 +1,235 @@
+    ) -> dict:
+        query_value = ""
+        if operation == "is":
+            query_value = " ".join((f'"{value}"' for value in value))
+        elif operation == "contains":
+            query_value = " OR ".join(value)
+            if len(value) > 1:
+                query_value = f"({query_value})"
+
+        elif operation == "regex":
+            query_value = " ".join((f"/{value}/" for value in value))
+        if asset_type == "all_fields":
+            query_string = f"{query_value}"
+        else:
+            query_string = f"{asset_type}:{query_value}"
+
+        if results_page_number:
+            return self._http_request(
+                "GET",
+                "search",
+                params={"query": query_string, "page": results_page_number},
+                auth=(self.email, self.api_key),
+                timeout=15,
+            )
+        else:
+            return self._http_request(
+                "GET",
+                "search",
+                params={"query": query_string},
+                auth=(self.email, self.api_key),
+            )
+
+
+def test_module(client: Client) -> str:
+    """
+    Returning 'ok' indicates that the integration works like it is supposed to. Connection to the service is successful.
+
+    Args:
+        client: HelloWorld client
+
+    Returns:
+        'ok' if test passed, anything else will fail the test.
+    """
+    try:
+        result = client.dehashed_search(
+            asset_type="vin", value=["test", "test1"], operation="is"
+        )
+    except Exception as e:
+        raise DemistoException(
+            f"Test failed. please check if Server Url, Email or Api key are correct. \n {e}"
+        )
+    else:
+        if isinstance(result, dict):
+            return "ok"
+        else:
+            return f"Test failed because got unexpected response from api: {result}"
+
+
+def convert_string_to_int(argument) -> int:
+    """
+    converts commands arguments to integers
+    :param argument: str
+    :return: converted argument as int
+    """
+    try:
+        input_as_int = int(argument)
+    except ValueError as e:
+        raise DemistoException(
+            f'"results_from" and "results_to expected" to be integers/n {e}'
+        )
+    else:
+        return input_as_int
+
+
+def filter_results(
+    entries: list, results_from: Union[str, None], results_to: Union[str, None]
+) -> tuple:
+    """
+    gets raw results returned from the api and limit the number of entries to return to demisto
+    :param entries: search results of the performed query
+    :param results_from: start range
+    :param results_to: end range
+    :return: filtered results
+    """
+    if results_from:
+        results_from_int = convert_string_to_int(results_from)
+    else:
+        results_from_int = RESULTS_FROM
+    if results_to:
+        results_to_int = convert_string_to_int(results_to)
+    else:
+        results_to_int = RESULTS_TO
+
```

Comment:
lets have a check here that `results_from_int` and `results_to_int` are valid.
that means that from < to, that they are not negative, and they are in the page.
and return the correct errors.